core/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -407,16 +408,14 @@ private void streamClosed() {
 
     private class InProcessServerStream implements ServerStream {
       final StatsTraceContext statsTraceCtx;
-      @GuardedBy("this")
       private ClientStreamListener clientStreamListener;
+      private final SynchronizationContext syncContext;
       @GuardedBy("this")
       private int clientRequested;
       @GuardedBy("this")
       private ArrayDeque<StreamListener.MessageProducer> clientReceiveQueue =
           new ArrayDeque<>();
-      @GuardedBy("this")
```

Comment:
Let's keep these under the lock, but save them to a local variable so they can be accessed from the callback.